.docker/node/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM node:10-alpine
+
+RUN mkdir -p /root/drupal-admin-ui
+WORKDIR /root/drupal-admin-ui
+
+ADD package.json /root/drupal-admin-ui/
+ADD yarn.lock /root/drupal-admin-ui/
+ADD packages /root/drupal-admin-ui/packages/
+
+RUN yarn install && \
```

Comment:
I can see a brittle aspect of this design... something that I found did not work without alteration.

this yarn install was failing with this error.

>info There appears to be trouble with your network connection. Retrying...

It is a combination of low compute power / low capacity network. and  rate limiting from a remote package repository.

stackoverflow has lots of discussion about how best to connect up a local repository package cache.

if 'production' then 
  do xyz
else 
  // in development pull from local/host  yarn package cache.

I am not going to comment further .. just to say it is tricky -- but I have a workaround I added a 

yarn install --network-timeout 1000000

This is a poor solution -- I dislike having to set an implementation detail - tuned  to the lowest common denominator  docker-machine environment ... but I also have to admit that  docker is not as run everywhere as people may think.

Reply:
Thank you for the quick response.

As someone who has read many the different approaches on  stack overflow ... the change is elegant and  a good step in the right direction.

I am looking into this and hope to report more over the weekend .. 

BUT it looks like I still need to add a network-timeout switch to my build as I see ESOCKETTIMEDOUT errors. It is always on the same file as before  "icons-3.0.1.tgz"

which makes me suspicious .. either a particularly troublesome server or something nasty like we are hitting a pathological corner case...   exceeding a internal cache-size limit causing the whole cache to be reset.  but enough wild speculation.

`
drupal_admin_ui_chrome | Starting ChromeDriver 2.38 (f91d32489882be7df38da3422a19713bfd113fa5) on port 9515
drupal_admin_ui_chrome | Remote connections are allowed by a whitelist (172.18.0.3).
drupal_admin_ui_node | yarn install v1.12.3
drupal_admin_ui_node | [1/5] Validating package.json...
drupal_admin_ui_node | [2/5] Resolving packages...
drupal_admin_ui_node | [3/5] Fetching packages...
drupal_admin_ui_drupal | ##############################################################################################
drupal_admin_ui_drupal | # One time login URL                                                                         #
drupal_admin_ui_drupal | # http://127.0.0.1/user/reset/1/1544709607/lAzMJIkPTceJR9dHHmx5Lj7Q6OCWJOpuhBJArEl4pss/login #
drupal_admin_ui_drupal | ##############################################################################################
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info If you think this is a bug, please open a bug report with the information provided in "/root/drupal-admin-ui/yarn-error.log".
drupal_admin_ui_node | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
drupal_admin_ui_node | error An unexpected error occurred: "https://registry.yarnpkg.com/@material-ui/icons/-/icons-3.0.1.tgz: ESOCKETTIMEDOUT".
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | info There appears to be trouble with your network connection. Retrying...
drupal_admin_ui_node | yarn workspace v1.12.3
drupal_admin_ui_node | yarn run v1.12.3
drupal_admin_ui_node | $ react-scripts build
drupal_admin_ui_node | /bin/sh: react-scripts: not found
drupal_admin_ui_node | error Command failed with exit code 127.
drupal_admin_ui_node | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
drupal_admin_ui_node | error Command failed.
drupal_admin_ui_node | Exit code: 1
drupal_admin_ui_node | Command: /usr/local/bin/node
drupal_admin_ui_node | Arguments: /opt/yarn-v1.12.3/lib/cli.js build
drupal_admin_ui_node | Directory: /root/drupal-admin-ui/packages/admin-ui
drupal_admin_ui_node | Output:
drupal_admin_ui_node | 
drupal_admin_ui_node | info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
drupal_admin_ui_node | yarn workspace v1.12.3
`